### PR TITLE
refactored to use flatlist in place deprecated listview, 

### PR DESCRIPTION
**Pull request by rollymaduk**
Hi I refactored a bit of the code to use flatlist in place of listview and changed the selected  value to return an object with the props {key,label} instead of just key:string

### CODE REVIEW

**Comment by mnlbox** [comment]
Any chance to see this in next release?

**Comment by effexts** [comment]
@hiddentao could you please look at this? Thanks in advance.

**Comment by mnlbox** [comment]
Any news?

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hiddentao) [inline .idea/misc.xml]: Please add this file to `.gitignore`

File: .idea/vcs.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (hiddentao) [inline .idea/vcs.xml]: Please add this file to `.gitignore`

File: package.json [hunk 3]
```diff
@@ -1,14 +1,15 @@
 {
   "name": "react-native-modal-filter-picker",
-  "version": "1.3.2",
+  "version": "1.3.3",
```
Comment (hiddentao) [inline package.json]: Please undo this version change. This is likely going to have to be a major release.

File: src/index.js [hunk 4]
```diff
@@ -1,28 +1,26 @@
...
+import { Modal, View, FlatList, TouchableOpacity, Text, TextInput } from 'react-native'
 
 
 import styles from './styles'
 
 
 export default class ModalFilterPicker extends Component {
   constructor (props, ctx) {
-    super(props, ctx)
+    super(props, ctx);
```
Comment (hiddentao) [inline src/index.js]: No ; please

File: src/index.js [hunk 5]
```diff
@@ -35,16 +33,18 @@ export default class ModalFilterPicker extends Component {
...
       modal,
-    } = this.props
+      visible,
+      onCancel,
+    } = this.props;
 
     const renderedTitle = (!title) ? null : (
       <Text style={titleTextStyle || styles.titleTextStyle}>{title}</Text>
-    )
+    );
```
Comment (hiddentao) [inline src/index.js]: No semicolon please

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
```
Comment (hiddentao) [inline .eslintrc.js]: I don't want semicolons at the line endings, Can we add that rule please

**Comment by hiddentao** [comment]
This requires some changes before it can be merged, plus it's out of date with the base branch.

**Comment by hiddentao** [comment]
Anyone up for doing it?

**Comment by mnlbox** [comment]
@rollymaduk can you complete this please?

**Comment by rollymaduk** [comment]
@hiddentao , @mnlbox I have made the necessary changes and adjustments as requested

File: package.json (hunk 3 above)
Comment (rollymaduk) [reply package.json]: done

Comment (rollymaduk) [reply package.json]:
```suggestion
  "version": "1.3.3",
```
done

File: src/index.js (hunk 4 above)
Comment (rollymaduk) [reply src/index.js]: fixed

**Comment by hiddentao** [comment]
@rollymaduk Thanks. Please remove the `.idea` folder completely from source control and resolve the remaining conflict with `index.js`.

**Comment by migueldaipre** [comment]
Please merge this.

**Comment by hiddentao** [comment]
There are numerous conflicts within `index.js` - can't merge this until that's cleared up.

**Comment by Thilagavathi1** [comment]
@hiddentao and @rollymaduk  Any update on this PR.

**Comment by jmhinric** [comment]
@rollymaduk any chance to fix the conflicts to get this PR in?  I'm trying to update an inherited app that uses this as a dependency and needs this PR- thanks!

**Comment by kayaui** [comment]
> refactor to use rn flatlist
ds: options 
to
**ds: props.options**

**Comment by mnlbox** [comment]
Interesting. Simple PR hold here for more than a year. I'm going to uninstall this package and start using new one.

**Comment by nihp** [comment]
I have changed the ListView to Flatlist but in onSelect method it returns the value as object. Can any one tell how can I achieve this. While onselect the selected item is return as object.  So the text is not able to display after selecting. 

```
renderOptionList = () => {
   const {
     noResultsText,
     listViewProps,
     keyboardShouldPersistTaps,
     keyExtractor,
   } = this.props

   const { ds } = this.state;
   if (!ds.length) {
     return (
     <FlatList
       data={ds}
         keyExtractor={keyExtractor||this.keyExtractor}
         {...listViewProps}
         renderItem={() => (
           <View style={styles.noResults}>
             <Text style={styles.noResultsText}>{noResultsText}</Text>
           </View>
         )}
       />
     )
   } else {
     return (
       <FlatList
         keyExtractor={keyExtractor||this.keyExtractor}
         {...listViewProps}
         data={ds}
         renderItem={this.renderOption}
       />
     )
   }
 };

 renderOption = ({item}) => {
   const {
     selectedOption,
     renderOption,
     optionTextStyle,
     selectedOptionTextStyle
   } = this.props;

   const { key, label } = item
   let style = styles.optionStyle;
   let textStyle = optionTextStyle||styles.optionTextStyle;
   if (key === selectedOption) {
     style = styles.selectedOptionStyle;
     textStyle = selectedOptionTextStyle ||styles.selectedOptionTextStyle
   }

   if (renderOption) {
     return renderOption(item, key === selectedOption)
   } else {
     return (
       <TouchableOpacity activeOpacity={0.7}
         style={style}
         onPress={() => this.props.onSelect(item)}
       >
         <Text style={textStyle}>{label}</Text>
       </TouchableOpacity>
     )
   }
 };
```

**Comment by ws7one** [comment]
> Interesting. Simple PR hold here for more than a year. I'm going to uninstall this package and start using new one.

@mnlbox Do u have a new one that you are using?

**Comment by ryanphung** [comment]
Just upgraded to React-native 0.60 and this component become unusable...

Seems like this project is no longer actively maintained. @ws7one @mnlbox have you found a good alternative?

**Comment by ryanphung** [comment]
@hiddentao since @rollymaduk appears to be inactive for about a year I forked his PR and made a few fixes in a new PR #40 . Please let me know if we can merge. As of now the component breaks in RN 0.60. Thanks!

**Comment by hiddentao** [comment]
Fixed in #40

**Pull request closed without merging**